content/building/build-versioning.md

```diff
@@ -46,3 +46,112 @@ set -x
+    APP_STORE_CONNECT_KEY_IDENTIFIER: Encrypted(...)
+    APP_STORE_CONNECT_PRIVATE_KEY: Encrypted(...)
+```
+
+- `APP_STORE_CONNECT_KEY_IDENTIFIER`
+
+  In **App Store Connect > Users and Access > Keys**, this is the **Key ID** of the key.
+
+- `APP_STORE_CONNECT_ISSUER_ID`
+
+  In **App Store Connect > Users and Access > Keys**, this is the **Issuer ID** displayed above the table of active keys.
+
+- `APP_STORE_CONNECT_PRIVATE_KEY`
+
+  This is the private API key downloaded from App Store Connect. Note that when encrypting files via UI, they will be base64 encoded and would have to be decoded during the build. Alternativey, you can encrypt the **contents** of the file and save the encrypted value to the environment variable.
+
+{{<notebox>}}
+Alternatively, each property can be specified in the `scripts` section of the YAML file as a command argument to programs with dedicated flags. See the details [here](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/app-store-connect/fetch-signing-files.md#--issuer-idissuer_id). In that case, the environment variables will be fallbacks for missing values in scripts.
+{{</notebox>}}
+
+Once you have the App Store Connect API access set with mentioned above environment variables, you can get the build number using the tool and set your incremented project version
+
+```bash
+export APP_STORE_CONNECT_PRIVATE_KEY=$(echo $APP_STORE_CONNECT_PRIVATE_KEY | base64 --decode) # if you encrypted the file itself, not its content
+LATEST_BUILD_NUMBER=app-store-connect get-latest-app-store-build-number '1234567890' # The argument is your application's Apple ID
+agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
+```
+
+To use the latest build number from Testflight use a similar command
+
+```bash
+export APP_STORE_CONNECT_PRIVATE_KEY=$(echo $APP_STORE_CONNECT_PRIVATE_KEY | base64 --decode) # if you encrypted the file itself, not its content
+LATEST_BUILD_NUMBER=app-store-connect get-latest-testflight-build-number '1234567890' # The argument is your application's Apple ID
+agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
+```
+
+{{<notebox>}}
+You can specify an optional argument `--app-store-version=APP_STORE_VERSION` to get the latest build number for a particular version of your application (`CFBundleShortVersionString`). Check the details for [get-latest-app-store-build-number](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/app-store-connect/get-latest-app-store-build-number.md#optional-arguments-for-action-get-latest-app-store-build-number) or for [get-latest-testflight-build-number](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/app-store-connect/get-latest-testflight-build-number.md#optional-arguments-for-action-get-latest-testflight-build-number).
+{{</notebox>}}
+
+Alternatively, if you use `YAML` configuration, you may just export the value to an environment variable and use it under your `CFBundleVersion` in `Info.plist`.
+
```

Comment:
I would add a new subsection `Increment build number in yaml` and have an example yaml excerpt about how to use it in the context of a script step.